substrate/client/transaction-pool/src/fork_aware_txpool/view.rs

```diff
@@ -0,0 +1,373 @@
+		&self,
+		tx_hash: ExtrinsicHash<ChainApi>,
+	) -> Watcher<ExtrinsicHash<ChainApi>, ExtrinsicHash<ChainApi>> {
+		self.pool.validated_pool().create_watcher(tx_hash)
+	}
+
+	/// Revalidates some part of transaction from the internal pool.
+	///
+	/// Intended to run from revalidation worker. Revlidation can be terminated by sending message
+	/// to the rx channel provided within `finish_revalidation_worker_channels`. Results are sent
+	/// back over tx channels and shall be applied in maintain thread.
+	pub(super) async fn revalidate_later(
+		&self,
+		finish_revalidation_worker_channels: FinishRevalidationWorkerChannels<ChainApi>,
+	) {
+		use sp_runtime::SaturatedConversion;
+
+		let FinishRevalidationWorkerChannels {
+			mut finish_revalidation_request_rx,
+			revalidation_result_tx,
+		} = finish_revalidation_worker_channels;
+
+		log::debug!(target:LOG_TARGET, "view::revalidate_later: at {} starting", self.at.hash);
+		let start = Instant::now();
+		let validated_pool = self.pool.validated_pool();
+		let api = validated_pool.api();
+
+		let batch: Vec<_> = validated_pool.ready().collect();
+		let batch_len = batch.len();
+
+		//todo: sort batch by revalidation timestamp | maybe not needed at all? xts will be getting
+		//out of the view...
+		//todo: revalidate future, remove if invalid.
+
+		let mut invalid_hashes = Vec::new();
+		let mut revalidated = HashMap::new();
+
+		let mut validation_results = vec![];
+		let mut batch_iter = batch.into_iter();
+		loop {
+			let mut should_break = false;
+			tokio::select! {
+				_ = finish_revalidation_request_rx.recv() => {
+					log::trace!(target: LOG_TARGET, "view::revalidate_later: finish revalidation request received at {}.", self.at.hash);
+					should_break = true;
+				}
+				_ = async {
+					if let Some(tx) = batch_iter.next() {
+						let validation_result = (api.validate_transaction(self.at.hash, tx.source, tx.data.clone()).await, tx.hash, tx);
+						validation_results.push(validation_result);
+					} else {
+						{
+							self.revalidation_worker_channels.lock().as_mut().map(|ch| ch.remove_sender());
+						}
+						should_break = true;
+					}
+				} => {}
+			}
+
+			if should_break {
+				break;
+			}
+		}
+
+		let revalidation_duration = start.elapsed();
+		self.metrics.report(|metrics| {
+			metrics.view_revalidation_duration.observe(revalidation_duration.as_secs_f64());
+		});
+		log::info!(
+			target:LOG_TARGET,
+			"view::revalidate_later: at {:?} count: {}/{} took {:?}",
+			self.at.hash,
+			validation_results.len(),
+			batch_len,
+			revalidation_duration
+		);
+		log_xt_debug!(data:tuple, target:LOG_TARGET, validation_results.iter().map(|x| (x.1, &x.0)), "[{:?}] view::revalidate_later result: {:?}");
+
+		for (validation_result, tx_hash, tx) in validation_results {
+			match validation_result {
+				Ok(Err(TransactionValidityError::Invalid(_))) => {
+					invalid_hashes.push(tx_hash);
+				},
+				Ok(Err(TransactionValidityError::Unknown(_))) => {},
```

Comment:
It could be correctly validate in future, it was copied from here: https://github.com/paritytech/polkadot-sdk/blob/7e7c33453eeb14f47c6c4d0f98cc982e485edc77/substrate/client/transaction-pool/src/revalidation.rs#L109-L118
I've added comment.